components/cdc/src/endpoint.rs

```diff
@@ -298,32 +308,83 @@ impl<T: 'static + RaftStoreRouter<E>, E: KvEngine> Endpoint<T, E> {
             pd_client,
             tso_worker,
             timer: SteadyTimer::default(),
-            scan_speed_limter: speed_limiter,
+            scan_speed_limiter: speed_limiter,
             max_scan_batch_bytes,
             max_scan_batch_size,
+            config: config.clone(),
             workers,
             scan_concurrency_semaphore,
             raft_router,
             engine: PhantomData,
             observer,
             store_meta,
             concurrency_manager,
-            min_ts_interval: cfg.min_ts_interval.0,
             min_resolved_ts: TimeStamp::max(),
             min_ts_region_id: 0,
             old_value_cache,
             resolved_region_count: 0,
             unresolved_region_count: 0,
             sink_memory_quota,
-            hibernate_regions_compatible: cfg.hibernate_regions_compatible,
             tikv_clients: Arc::new(Mutex::new(HashMap::default())),
         };
         ep.register_min_ts_event();
         ep
     }
 
-    pub fn set_min_ts_interval(&mut self, dur: Duration) {
-        self.min_ts_interval = dur;
+    fn on_change_cfg(&mut self, mut change: ConfigChange) {
+        info!(
+            "cdc config updated";
+            "change" => ?change
+        );
+
+        // Check if min_ts_interval is 0 and incremental_scan_concurrency less than incremental_scan_threads.
+        // This is a double check, as we have already done this in the CdcConfig validate.
+        // But to prevent bugs caused by the validate change, so we check again.
+        if let Some(min_ts_interval) = change.get("min_ts_interval") {
+            if ReadableDuration::from(min_ts_interval.clone()).is_zero() {
+                change.remove("min_ts_interval");
+            }
+        }
+        if let Some(incremental_scan_concurrency) = change.get("incremental_scan_concurrency") {
+            if usize::from(incremental_scan_concurrency.clone())
+                < self.config.incremental_scan_threads
+            {
+                change.remove("incremental_scan_concurrency");
+            }
+        }
```

Comment:
```suggestion
        // Check if min_ts_interval is 0 and incremental_scan_concurrency less than incremental_scan_threads.
        // This is a double check, as we have already done this in the CdcConfig validate.
        // But to prevent bugs caused by the validate change, so we check again.
        let mut validate_cfg = self.config.clone();
        validate_cfg.update(change.clone());
        if let Err(e) = validate_cfg.validate() {
            warn!("some warnings"; "error" => ?e);
            return
        }

        info!(
            "cdc config updated";
            "change" => ?change
        );
```